wherobots/db/connection.py

```diff
@@ -230,7 +238,20 @@ def __request_results(self, execution_id: str) -> None:
         self.__send(request)
 
     def __cancel_query(self, execution_id: str) -> None:
-        query = self.__queries.pop(execution_id)
-        if query:
-            logging.info("Cancelled query %s.", execution_id)
-            # TODO: when protocol supports it, send cancellation request.
+        """Cancels the query with the given execution ID."""
+        if self.__protocol < MIN_VERSION_FOR_CANCEL:
```

Comment:
nit: should we be doing a semver aware comparison here? i.e. would this do the wrong thing comparing "1.2.0" and "1.10.0"?

Reply:
@sfishel18 `Version` objects implement comparison operators: https://packaging.pypa.io/en/stable/version.html#packaging.version.Version

Reply:
oh that's handy, wish javascript could have nice things like that :)